tests/test_csr.py

```diff
@@ -0,0 +1,139 @@
+from django.http import FileResponse
+from django.urls import reverse
+
+import pytest
+from cryptography import x509
+from cryptography.hazmat.backends import default_backend
+
+from simple_certmanager.models import SigningRequest
+
+
+@pytest.fixture
+def signing_request():
+    return SigningRequest(common_name="test.com")
+
+
+@pytest.mark.django_db
+def test_admin_create_signing_request(admin_client, db):
+    add_url = reverse("admin:simple_certmanager_signingrequest_add")
+
+    data = {
+        "common_name": "test.com",
+        "country_name": "US",
+        "organization_name": "Test Org",
+        "state_or_province_name": "Test State",
+        "email_address": "test@test.com",
+    }
+
+    response = admin_client.post(add_url, data, follow=True)
+
+    assert response.status_code == 200
+
+    signing_request = SigningRequest.objects.get()
+    assert signing_request is not None
+    assert signing_request.private_key != ""
+    assert signing_request.csr != ""
+    assert "BEGIN PRIVATE KEY" in signing_request.private_key
+
+
+@pytest.mark.django_db
+def test_save_generates_private_key_and_csr(signing_request):
+    assert signing_request.private_key == ""
+    assert signing_request.csr == ""
+    signing_request.save()
+    saved_private_key = signing_request.private_key
+    assert signing_request.private_key != ""
+    assert signing_request.csr != ""
+    # Additional saves do not overwrite the private key
+    signing_request.save()
+    assert signing_request.private_key == saved_private_key
+
+
+@pytest.mark.django_db
+def test_generate_csr():
+    signing_request = SigningRequest(
+        common_name="test.com",
+        country_name="US",
+        organization_name="Test Org",
+        state_or_province_name="Test State",
+        email_address="test@test.com",
+    )
+
+    signing_request.generate_private_key()
+    signing_request.save()
```

Comment:
```suggestion
    signing_request = SigningRequest.objects.create(
        common_name="test.com",
        country_name="US",
        organization_name="Test Org",
        state_or_province_name="Test State",
        email_address="test@test.com",
    )
```
should be equivalent, since your `def save` takes care of generating the privatey key and CSR.

It was confusing me why the explicit call to generate a private key and additional save call were needed.